SerialPrograms/Source/Integrations/DppIntegration/DppClient.cpp

```diff
@@ -152,11 +151,14 @@ void Client::run(const std::string& token) {
     try {
         uint32_t intents = intents::i_default_intents | intents::i_guild_members;
         m_bot = std::make_unique<cluster>(token, intents);
```

Comment:
This still has the problem of initializing asynchronously from the starting thread. Likewise, if `connect()` is called very quickly twice in the row before the thread can start up, it will crash.

Is it possible to move this line `m_bot = std::make_unique<cluster>(token, intents);` to line 37 (just before the thread is started)?

This also protects against multiple `connect()` calls since the first one will get in and construct it. Then the later ones will see that is has already been constructed and will exit.

Reply:
It's possible for the `start` button to be clicked in rapid succession? Yeah, it can definitely be moved.

Reply:
Doesn't necessarily have to be that quick. Could also be CPU starved such at the new thread takes forever to startup. At this is the obvious example of how it can get called twice. There might be more involving how it's auto-started with the program.

Either way, better to make it robust to this since someone may call it differently in the future. lol